### PR TITLE
Fixed URI parsing with 2 #'s

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

**Comment by andrewconner** [comment]
This was a Java URI parsing problem. @eishay @ymatsuda

File: shoebox/app/com/keepit/common/net/URI.scala [hunk 1]
```diff
@@ -49,11 +52,15 @@ object URI extends Logging {
 
   val twoHexDigits = """\p{XDigit}\p{XDigit}""".r
   val encodedPercent = java.net.URLEncoder.encode("%", "UTF-8")
+  val encodedHash = java.net.URLEncoder.encode("%", "UTF-8")
```
Comment (ymatsuda) [inline shoebox/app/com/keepit/common/net/URI.scala:55]: This should be encode("#", "UTF-8")

File: shoebox/app/com/keepit/common/net/URI.scala [hunk 2]
```diff
@@ -49,11 +52,15 @@ object URI extends Logging {
...
 
   def fixMalformedEscape(uriString: String) = {
-    uriString.split("%", -1) match {
+    (uriString.split("%", -1) match {
       case Array(first, rest @ _*) =>
         rest.foldLeft(first){ (str, piece) => str + twoHexDigits.findPrefixOf(piece).map(_ => "%").getOrElse(encodedPercent) + piece }
+    }).split("#",-1) match {
+      case Array(first, rest @ _*) =>
+        rest.foldLeft(first){ (str, piece) => str + twoHexDigits.findPrefixOf(piece).map(_ => "#").getOrElse(encodedHash) + piece }
```
Comment (ymatsuda) [inline shoebox/app/com/keepit/common/net/URI.scala:63]: we can't replicate the code for problematic % signs for processing #. that one is to fix % signs not followed by hexdigits.
we should output the first # as it is, and others, if any, should be output as escapedHash.

**Comment by 2is10** [comment]
This kind of fix should definitely have tests.

**Comment by andrewconner** [comment]
@2is10 Agreed. Incoming, wanted to get the fix pushed. We currently don't have any URI tests :(

**Pull request merged**